zendesk/zendesk_client_service.go

```diff
@@ -275,6 +275,69 @@ func (c *client) getAll(endpoint string, in interface{}) ([]Ticket, error) {
 	return result, err
 }
 
+func (c *client) getOneByOne(in interface{}) ([]Ticket, error) {
+	endpointPrefix := "/api/v2/tickets/"
+	endpointPostfix := ".json"
+	result := make([]Ticket, 0)
+	payload, err := marshall(in)
+	if err != nil {
+		return nil, err
+	}
+
+	headers := map[string]string{}
+	if in != nil {
+		headers["Content-Type"] = "application/json"
+	}
+	dataPerPage := new(APIPayload)
+
+	// currently we can manually set the starting and ending IDs for data pulling
+	// because memory may reach its limit if the dataset is too large
+	// ideally, we want to load data to database in batches on the fly
+	// instead of loading the entire chunk
+	startID := 1
+	endID := 10000
```

Comment:
is the endID always 10000, what if the number of records increase?

Reply:
I explained that in the comments above the line. Currently we need to manually adjust the numbers due to the memory limit